### PR TITLE
Include rubyisms

### PR DESCRIPTION
**Pull request by peterwilson**
@ColinDKelley @alecjacobs5401 Took a pass at updating our style guide a bit to include 1) it's intention (now that we have rubocop), 2) how to propose changes, 3) a snippet about `!!` and when to/not to use it. Not guaranteeing this is perfect but it's a first pass!

I was also considering splitting this out into multiple `.md` files, one for style and one for the Ruby-isms. Style is visual, but Ruby-isms can be more impactful to logic or intent. Having them on different pages may help make that clearer. But I decided to not clutter this PR with that code movement, just the code change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,37 @@
 # Prelude
 Invoca Ruby Style Guide initially forked from https://github.com/bbatsov/ruby-style-guide.
 
-## Table of Contents
+Our Ruby Style Guide is enforced by [Rubocop](https://github.com/bbatsov/rubocop), a static code analyzer.
+
+### Making Changes
+
+This guide exists to represent the collective knowledge and agreed upon standards Invoca's of Engineering department. Changes are welcome and encouraged from all sources! Contributors should try, and may be required, to provide background information and/or reasoning behind their proposed change.
```
Comment (alecjacobs5401) [inline README.md]: > standards **of Invoca's** Engineering

File: README.md [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
+  * [Regular Expressions](#regular-expressions)
+  * [Metaprogramming](#metaprogramming)
+  * [Misc](#misc)
+* [Preferred Ruby-isms](#preferred-ruby-isms)
+  * [Casting Booleans](#casting-booleans)
+
 
 ## Source Code Layout
 
+Items under this section should be primarily focused on code style. These types of topics should almost always be caught by Rubocop. Additions and deletions to this section likely will require an update to our `.rubocop.yml` templates.
```
Comment (alecjacobs5401) [inline README.md]: could we link to the master web `.rubocop.yml` file here?

File: README.md [hunk 3]
```diff
@@ -1112,3 +1126,48 @@ Ideally, most methods will be shorter than 5 lines of code. Comments and empty l
...
+```ruby
+# JSON expects true/false, not truthiness
+def to_json
+  record = find_by_id(1)
+  {
+    record_exists: !!record
+  }
+end
+
+# Ruby convention says query methods, those ending in `?`, should return a true or false value
```
Comment (ColinDKelley) [inline README.md]: Not true! Ruby convention is that `?` methods are predicates, and predicates return truthiness. Even the standard library has `nonblank?`. And tons of methods in the standard library and all over our code base return `nil` as a falsey value sometimes in addition to `false`. Every predicate that uses the Hobo operator (soon to be lonely operator when we're on Ruby 2.4) is this way.

We definitely do not want to sprinkle `!!` all over our code base, which is what this implies.

We only need `!!` when we need to force truthiness to `true`/`false`.

I think you could move this example into the unnecessary section.

File: README.md (hunk 2 above)
Comment (ColinDKelley) [reply README.md]: Great idea.

Comment (peterwilson) [reply README.md]: This is a public repo, do we want to be linking to our private repos?

Comment (alecjacobs5401) [reply README.md]: oops, totally forgot about that

File: README.md (hunk 3 above)
Comment (peterwilson) [reply README.md]: `nonblank?` isn't part of the standard library, is it? It looks like it's a method we wrote as an extension to Hobo, and we deprecated it in favor of the Rails method `presence`.

I'd concede to add nil, and have true/false/nil as acceptable values. But I'm not aware of standard library methods, or largely accepted public libraries, that use predicate methods to return non true/false/nil values.

Comment (ColinDKelley) [reply README.md]: Oops, scratch 'nonblank?`. I meant `nonzero?`.

Comment (ColinDKelley) [reply README.md]: https://stackoverflow.com/questions/10527997/should-a-method-ending-in-question-mark-return-only-a-boolean

Comment (peterwilson) [reply README.md]: Huh, weird. Feels like that pattern doesn't follow the principle of least surprise, but that definitely is the standard library.

I'll remove that comment/example from the `good` section.

**Comment by peterwilson** [comment]
Closing in favor of https://github.com/Invoca/style-guide/pull/8

**Pull request closed without merging**